### ISSUE TITLE
Max idle timeout is perhaps too short

### ISSUE THREAD

**Issue by msoxzw** [mozilla/neqo#1412]
The maximum idle timeout in neqo used by Firefox is [30 seonds](https://github.com/mozilla/neqo/blob/07c2019988a8f0a37f87cbd90f95e906e7b53258/neqo-transport/src/connection/params.rs#L29), eclipsed by [600 seconds](
https://github.com/google/quiche/blob/1fdac9bdbf85d27537e8ce47ca4f1eaa034f1ff5/quiche/quic/core/quic_constants.h#L152) in quiche used by Chrome.

Considering remote servers, https://www.cloudflare.com/ is 180 seconds, and https://www.google.com/ is 240 seconds.

I reckon that neqo probably has not implemented the QUIC handshake timeout, and as a result 30 seconds idle timeout is picked and chosen.

**Comment by martinthomson** [comment]
Yes, our timeouts are a little shorter than you will see elsewhere.  We have a single timeout for the connection, regardless of its state; handshakes need to fit inside that time.  Idle timeouts also drive our keep-alive behaviour in cases where we want to keep a connection open (we do this if the HTTP client has an outstanding request).

The value is configurable, we just haven't seen much reason to reassess the 30s value.  @msoxzw do you have a reason that this should be longer?

**Comment by msoxzw** [comment]
Longer idle timeout tends to avoid unnecessary expensive handshakes. If QUIC handshake timeout is used like Google quiche, ngtcp2, etc. , an outstanding request could be closed in a much shorter time, e.g. 10 seconds.
